Observer/RecommendSettings.php

```diff
@@ -154,12 +159,46 @@ protected function validateRecommendation(string $changedPath, string $recommend
             throw new LocalizedException(__(
                 "Unable to save %1 Recommend configuration due to the following error: %2",
                     $modelName,
-                    $e->getMessage()
+                    $this->getUserFriendlyRecommendApiErrorMessage($e)
                 )
             );
         }
     }
 
+    /**
+     * If API does not return a hits response the model may not be configured correctly.
+     * Do not hard fail but alert the end user.
+     * @throws LocalizedException
+     */
+    protected function shouldDisplayWarning(array $recommendationResponse): bool
+    {
+        return
+            $this->appState->getAreaCode() === \Magento\Framework\App\Area::AREA_ADMINHTML
```

Comment:
Why do we need to check this ? Any reasons why the area code could be different ?

Reply:
The event observed by this class is dispatched by `Magento\Config\Model\Config::save` so it's possible it might not be triggered from the admin panel (perhaps CLI scripts, web API's, other customizations etc.). I wanted to play it safe so as not to introduce any unwanted side effects.

Reply:
ok makes sense :)